docs/source/semantic_segmentation.mdx

```diff
@@ -0,0 +1,161 @@
+# Image classification
+
+Semantic segmentation datasets are used to train a model to classify every pixel in an image. There are
+a wide variety of applications enabled by these datasets such as background removal from images and stylizing
+them or scene understanding for autonomous driving. This guide will show you how to apply transformations
+to an image classification dataset.
+
+Before you start, make sure you have up-to-date versions of `albumentations` and `cv2` installed:
+
+```bash
+pip install -U albumentations opencv-python
+```
+
+This guide uses the [Scene Parsing](https://huggingface.co/datasets/scene_parse_150) dataset for segmenting
+and parsing an image into different image regions associated with semantic categories, such as sky, road, person, and bed.
+
+Load the dataset and take a look at an example:
+
+```py
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("scene_parse_150")
+>>> index = 10
+>>> dataset["train"][index]
+{'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=683x512 at 0x7FB37B0EC810>,
+ 'annotation': <PIL.PngImagePlugin.PngImageFile image mode=L size=683x512 at 0x7FB37B0EC9D0>,
+ 'scene_category': 927}
+```
+
+The dataset has three fields:
+
+* `image`: a PIL image object.
+* `annotation`: segmentation mask of the image.
+* `scene_category`: the label or scene category of the image (like “kitchen” or “office”).
+
+Next, check out an image with `dataset["train"][index]["image"]`:
+
+<div class="flex justify-center">
+    <img src="https://i.ibb.co/NKYHp5v/download.png">
+</div>
+
+Similarly, you can check out the respective segmentation mask (`dataset["train"][index]["annotation"]`):  
+
+<div class="flex justify-center">
+    <img src="https://i.imgur.com/qhTK2at.png">
+</div>
+
+We can also overlay the segmentation mask on top of the original image as is commonly done
+when visualizing segmentation datasets. For that, we first need to define
+a [color palette](https://github.com/tensorflow/models/blob/3f1ca33afe3c1631b733ea7e40c294273b9e406d/research/deeplab/utils/get_dataset_colormap.py#L51):
+
+After defining the color palette, you should be ready to visualize some overlays. 
+
+```py
+>>> import matplotlib.pyplot as plt
+
+>>> def visualize_seg_mask(image: np.ndarray, mask: np.ndarray):
+...    color_seg = np.zeros((mask.shape[0], mask.shape[1], 3), dtype=np.uint8)
+...    palette = np.array(create_ade20k_label_colormap())
```

Comment:
Maybe actually define this function instead of linking to it above and never defining it?

Reply:
idk fine either way I suppose. It's quite long

Reply:
Yeah, because of the length, I decided not to include it.